frontend/src/routes/Search.tsx

```diff
@@ -114,17 +164,162 @@ const SearchPage: React.FC<Props> = ({ q, outcome }) => {
+
+    const isActive = startDate || endDate;
+    const inputStyle = {
+        borderRadius: 4,
+        border: `1px solid ${COLORS.neutral40}`,
+        ...focusStyle({ width: 2, inset: true }),
+    };
+
+    return <FloatingContainer
+        {...{ ref }}
+        placement="top"
+        arrowSize={12}
+        ariaRole="dialog"
+        trigger="click"
+        viewPortMargin={12}
+    >
+        <FloatingTrigger>
+            <Button aria-label={t("search.select-time-frame")} css={{
+                height: 40,
+                ...isActive && {
+                    backgroundColor: COLORS.neutral30,
+                    "&&": {
+                        border: `1px solid ${COLORS.neutral90}`,
+                    },
+                },
+            }}><LuCalendarRange /></Button>
+        </FloatingTrigger>
+        <Floating css={{ padding: "0 8px 8px 8px" }}>
+            <p css={{ fontSize: 14, padding: "4px 2px" }}>{t("search.select-time-frame")}</p>
+            <div css={{
+                display: "flex",
+                flexWrap: "wrap",
+                gap: 8,
+            }}>
+                {isActive && <ProtoButton
+                    aria-label={t("search.clear-time-frame")}
+                    css={{ display: "flex", alignItems: "center" }}
+                    onClick={() => {
+                        handleChange("", "start");
+                        handleChange("", "end");
+                    }}
+                ><LuX /></ProtoButton>}
+                <input
+                    value={startDate ?? ""}
+                    css={inputStyle}
+                    type="date"
+                    onChange={e => handleChange(e.target.value, "start")}
+                />
+                <span>-</span>
+                <input
+                    value={endDate ?? ""}
+                    css={inputStyle}
+                    type="date"
+                    min={startDate ?? ""}
+                    onChange={e => handleChange(e.target.value, "end")}
+                />
+            </div>
+        </Floating>
+    </FloatingContainer>;
+};
+
+type FilterButtonProps = {
+    type?: Item;
+    router: RouterControl;
+}
+
+const FilterButton: React.FC<FilterButtonProps> = ({ type, router }) => {
+    const { t } = useTranslation();
+    const handleClick = () => {
+        const params = new URLSearchParams(window.location.search);
+        const query = params.get("q") ?? "";
+        const start = params.get("start") ?? "";
+        const end = params.get("end") ?? "";
+        const itemType = type ?? undefined;
+        router.goto(SearchRoute.url({ query, itemType, start, end }), true);
+    };
+
+    const params = new URLSearchParams(window.location.search);
+    const filter = params.get("f");
+    const translationKey = type ? type.toLowerCase() as "event" | "realm" : "all";
+
+    return <Button
+        onClick={handleClick}
+        disabled={!filter && !type || filter === type}
```

Comment:
That's slightly wrong. When someone adds `f=banana` then `!filter` is false. You want to wrap the `params.get("f")` above in `validType()` again, so that you also get `null` for invalid values.